frontend/js/src/user/stats/components/Choropleth.tsx

```diff
@@ -176,66 +180,168 @@ export default function CustomChoropleth(props: ChoroplethProps) {
+          <button
+            type="button"
+            className="btn btn-info btn-rounded btn-sm"
+            title={`Play tracks from ${countryName}`}
+            style={{
+              borderRadius: "50%",
+              width: "30px",
+              height: "30px",
+              display: "flex",
+              alignItems: "center",
+              justifyContent: "center",
+              border: "none",
+              cursor: "pointer",
+              background: "#353070",
+              color: "white",
+            }}
+            onClick={(e) => {
+              e.stopPropagation();
+              handlePlayCountryTracks(countryName);
+            }}
+          >
+            <FontAwesomeIcon icon={faPlayCircle as IconProp} />
+          </button>
         </div>
-        <hr style={{ margin: "0.5em 0" }} />
-        {artists?.slice(0, 10).map((artist: UserArtistMapArtist) => (
-          <div key={artist.artist_mbid}>
-            <span className="badge color-purple" style={{ marginRight: "4px" }}>
-              <FontAwesomeIcon
-                style={{ marginRight: "4px" }}
-                icon={faHeadphones as IconProp}
-              />
-              {artist.listen_count}
+        <div style={{ padding: "8px 12px" }}>
+          <div
+            style={{
+              display: "flex",
+              alignItems: "center",
+            }}
+          >
+            <Chip color={selectedCountry.color!} style={{ marginRight: 7 }} />
+            <span>
+              {"My Listens: "}
+              <strong>
+                {value} {suffix}
+              </strong>
             </span>
-            <Link to={`/artist/${artist.artist_mbid}/`}>
-              {artist.artist_name}
-            </Link>
-            <br />
           </div>
-        ))}
-      </div>
+
+          {artists?.length > 0 && (
+            <>
+              <hr style={{ margin: "0.5em 0" }} />
+              <div
+                style={{
+                  maxHeight: "200px",
+                  overflowY: "auto",
+                  overflowX: "hidden",
+                }}
+              >
+                {artists?.map((artist: UserArtistMapArtist) => (
+                  <div key={artist.artist_mbid}>
+                    <span
+                      className="badge color-purple"
+                      style={{ marginRight: "4px" }}
+                    >
+                      <FontAwesomeIcon
+                        style={{ marginRight: "4px" }}
+                        icon={faHeadphones as IconProp}
+                      />
+                      {artist.listen_count}
+                    </span>
+                    <Link to={`/artist/${artist.artist_mbid}/`}>
+                      {artist.artist_name}
+                    </Link>
+                    <br />
+                  </div>
+                ))}
+              </div>
+            </>
+          )}
+        </div>
+      </button>
     );
   }, [selectedCountry, selectedMetric]);
```

Comment:
I see an eslint warning for this line; you need to add `handlePlayCountryTracks` as a dependency of this hook